pypeit/spectrographs/p200_tspec.py

```diff
@@ -0,0 +1,269 @@
+""" Module for Keck/NIRES specific codes
+"""
+import numpy as np
+from astropy.time import Time
+
+from pypeit import msgs
+from pypeit import telescopes
+from pypeit.core import framematch
+from pypeit.par import pypeitpar
+from pypeit.spectrographs import spectrograph
+from pypeit.images import detector_container
+
+from pkg_resources import resource_filename
+
+
+class P200TSPECSpectrograph(spectrograph.Spectrograph):
+    """
+    Child to handle Keck/NIRES specific code
+    """
+    ndet = 1
+
+    def __init__(self):
+        # Get it started
+        super(P200TSPECSpectrograph, self).__init__()
+        self.spectrograph = 'p200_tspec'
+        self.telescope = telescopes.KeckTelescopePar()
+        self.camera = 'TSPEC'
+        self.numhead = 1
+
+    def init_meta(self):
+        """
+        Generate the meta data dict
+        Note that the children can add to this
+
+        Returns:
+            self.meta: dict (generated in place)
+
+        """
+        meta = {}
+        # Required (core)
+        meta['ra'] = dict(ext=0, card='RA')
+        meta['dec'] = dict(ext=0, card='DEC')
+        meta['target'] = dict(ext=0, card='OBJECT')
+        meta['decker'] = dict(ext=0, card=None, default='default')
+        meta['binning'] = dict(ext=0, card=None, default='1,1')
+
+        meta['mjd'] = dict(ext=0, card=None, compound=True)
+        meta['exptime'] = dict(ext=0, card='EXPTIME')
+        meta['airmass'] = dict(ext=0, card='AIRMASS')
+        # Extras for config and frametyping
+        meta['dispname'] = dict(ext=0, card='FPA')
+        meta['idname'] = dict(ext=0, card='OBSTYPE')
+
+        # Ingest
+        self.meta = meta
+
+    def compound_meta(self, headarr, meta_key):
+        if meta_key == 'mjd':
+            time = headarr[0]['UTSHUT']
+            ttime = Time(time, format='isot')
+            return ttime.mjd
+        else:
+            msgs.error("Not ready for this compound meta")
+
+    def get_detector_par(self, hdu, det):
+        # Detector 1
+        detector_dict = dict(
+            binning='1,1',
+            det=1,
+            dataext         = 0,
+            specaxis        = 1,
+            specflip        = True,
+            spatflip=False,
+            platescale      = 0.37,
+            darkcurr        = 0.085,
+            saturation      = 28000,
+            nonlinear       = 0.9,
+            mincounts       = -1e10,
+            numamplifiers   = 1,
+            gain            = np.atleast_1d(3.8),
+            ronoise         = np.atleast_1d(3.5),
+            datasec         = np.atleast_1d('[:,:]'),
+            oscansec        = np.atleast_1d('[:,:]')
+            )
+        detector = detector_container.DetectorContainer(**detector_dict)
+        return detector
+
+    @property
+    def pypeline(self):
+        return 'Echelle'
+
```

Comment:
If static, i.e. if this does not depend on self, then we have been making them @classmethod, see other spectrograph file examples.